PluginCore/PluginCore/FRService/FRSearch.cs

```diff
@@ -608,6 +612,61 @@ private List<SearchMatch> SearchSource(string src, int startIndex, int startLine
             return results;
         }
 
+        private bool LookupRegex(string ba, ref int i)
+        {
+            if (SourceFile == null)
+                return false;
+
+            string ext = System.IO.Path.GetExtension(SourceFile).ToLowerInvariant();
+
+            // Haxe, ActionScript, and JavaScript support Regex literals
+            if (!Helpers.FileInspector.IsActionScript(SourceFile, ext) &&
```

Comment:
But Scintilla syntax configuration also depends on file extensions, plus the Haxe configuration file doesn't have the entry set (I guess because since regexes start with ~ they aren't styled correctly).

I prefer how it is currently than assume all files equal. If PluginCore had access to the language syntax files maybe that could be used as you say, although I'd go for individual check for Haxe at any rate. IMHO, I'd leave it this way for now. If in the future FRSearch needs further language specific checks I'd revisit then how to handle it.